app/src/main/java/me/eugeniomarletti/renderthread/CanvasUtils.java

```diff
@@ -0,0 +1,26 @@
+package me.eugeniomarletti.renderthread;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.os.Build;
+import android.support.annotation.NonNull;
+
+public final class CanvasUtils {
+
+    private CanvasUtils() {
+    }
+
+    private static final RectF TMP_RECT = Build.VERSION.SDK_INT < 21 ? new RectF() : null;
+
+    public static void drawRoundRect(
+            @NonNull Canvas canvas, float left, float top, float right, float bottom, float rx, float ry, @NonNull Paint paint) {
+
+        if (Build.VERSION.SDK_INT < 21) {
```

Comment:
Why don't check if TMP_RECT is null? Or extract the SDK version in a method? (it improves the readability, I think)